### PR TITLE
Remove scss lang directives from Vue SFC style tags; add eslint rule

### LINKED ISSUE #1087: Convert components using `lang="scss"` style blocks to plain css
Many of our components use Sass in their style blocks instead of plain CSS. These should be converted to Tailwind where possible and scss should be eliminated. Many components with `<style lang="scss">` aren't even using Sass features. 

This reduces reliance on yet another technology in our toolchain, encourages better practices (if you feel like you need a loop/control structure in your css, you should probably write a Vue component instead of whatever logic you're writing), and, down the line, as part of the [Remove legacy styles](https://github.com/WordPress/openverse-frontend/milestone/13) milestone, would let us remove around 6mb of dev dependencies, `sass` and `sass-loader`.

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #1087 by @zackkrida 

## Description

Converts all Vue SFCs to plain css in the `<style>` tags. Also adds an eslint rule to prevent contributors from using scss or any lang in Vue SFCs.

## Testing Instructions

- `pnpm i` and try to add a `lang` to a style block in a Vue SFC
- Observe that modified pages look correct/consistent with their original appearances.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by krysal — APPROVED** [review]
LGTM!

File: .eslintrc.js [hunk 1]
```diff
@@ -44,6 +44,12 @@ module.exports = {
     'vue/html-closing-bracket-newline': 'off',
     'vue/html-indent': 'off',
     'vue/singleline-html-element-content-newline': 'off',
+    'vue/block-lang': [
+      'error',
+      {
+        style: { allowNoLang: true },
```
[the remark this answers is not in the document]
Comment (zackkrida) [reply .eslintrc.js:50]: I agree!

Comment (zackkrida) [inline .eslintrc.js:50]:
```suggestion
	// This confusing naming prevents the use of 'lang' directives 
	// entirely on Vue SFC style blocks.
        style: { allowNoLang: true },
```

File: .eslintrc.js [hunk 2]
```diff
@@ -44,6 +44,14 @@ module.exports = {
     'vue/html-closing-bracket-newline': 'off',
     'vue/html-indent': 'off',
     'vue/singleline-html-element-content-newline': 'off',
+    'vue/block-lang': [
+      'error',
+      {
+	// This confusing naming prevents the use of 'lang' directives 
+	// entirely on Vue SFC style blocks.
```
Comment (zackkrida) [inline .eslintrc.js:51]:
```suggestion
            // This confusing naming prevents the use of 'lang' directives 
            // entirely on Vue SFC style blocks.
```

File: .eslintrc.js [hunk 3]
```diff
@@ -44,6 +44,14 @@ module.exports = {
     'vue/html-closing-bracket-newline': 'off',
     'vue/html-indent': 'off',
     'vue/singleline-html-element-content-newline': 'off',
+    'vue/block-lang': [
+      'error',
+      {
+            // This confusing naming prevents the use of 'lang' directives 
+            // entirely on Vue SFC style blocks.
```
Comment (zackkrida) [inline .eslintrc.js:51]:
```suggestion
          // This confusing naming prevents the use of 'lang' directives 
          // entirely on Vue SFC style blocks.
```

File: .eslintrc.js [hunk 4]
```diff
@@ -44,6 +44,14 @@ module.exports = {
     'vue/html-closing-bracket-newline': 'off',
     'vue/html-indent': 'off',
     'vue/singleline-html-element-content-newline': 'off',
+    'vue/block-lang': [
+      'error',
+      {
+          // This confusing naming prevents the use of 'lang' directives 
+          // entirely on Vue SFC style blocks.
```
Comment (zackkrida) [inline .eslintrc.js:51]:
```suggestion
        // This confusing naming prevents the use of 'lang' directives 
        // entirely on Vue SFC style blocks.
```

File: .eslintrc.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (zackkrida) [reply .eslintrc.js:52]: @krysal I added a comment to make this clearer, hopefully.

File: src/locales/po-files/openverse.pot [hunk 5]
```diff
@@ -1868,7 +1868,6 @@ msgctxt "hero.aria.search-type"
 msgid "search type"
 msgstr ""
 
-#: src/components/LocaleSelector.vue:3
 msgctxt "hero.locale.label"
```
Comment (obulat) [inline src/locales/po-files/openverse.pot:1871]: The locale selector strings are still there in the `.pot` file. I guess if we delete them from the JSON file, they will be removed from `.pot`, too?

**Review by obulat — APPROVED** [review]
I've added a couple of change requests in form of a PR: #1201. Other than that, everything LGTM! I thought this would require many more changes, especially around tabs.

**Pull request merged**